# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/WindowDestroyedEvent.cpp
# Commit: c63ecdad6a3c
# Full Hash: c63ecdad6a3c60ba625517d34bb29c8aae9f5e95
# Author: Henri Sivonen <hsivonen@hsivonen.fi>
# Date: 2021-02-22 17:34:52
# Regressor Bug: 1663931
# File Overlap Count: 1
# Description:
#   Bug 1663931 - Avoid moving focus when changing iframe remoteness. r=nika,mccr8
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D99215
# ==============================================================================

diff -r 828c62bfc0a2 -r c63ecdad6a3c dom/base/WindowDestroyedEvent.cpp
--- a/dom/base/WindowDestroyedEvent.cpp	Mon Feb 22 10:48:53 2021 +0000
+++ b/dom/base/WindowDestroyedEvent.cpp	Mon Feb 22 10:51:51 2021 +0000
@@ -19,6 +19,7 @@
 #include "mozilla/BasePrincipal.h"
 #include "mozilla/Components.h"
 #include "mozilla/ProfilerLabels.h"
+#include "nsFocusManager.h"
 
 namespace mozilla {
 
@@ -52,6 +53,8 @@
 WindowDestroyedEvent::Run() {
   AUTO_PROFILER_LABEL("WindowDestroyedEvent::Run", OTHER);
 
+  nsCOMPtr<nsPIDOMWindowOuter> nukedOuter;
+
   nsCOMPtr<nsIObserverService> observerService = services::GetObserverService();
   if (!observerService) {
     return NS_OK;
@@ -105,6 +108,7 @@
           nsGlobalWindowOuter* outer =
               nsGlobalWindowOuter::FromSupports(window);
           currentInner = outer->GetCurrentInnerWindowInternal();
+          nukedOuter = outer;
         }
         NS_ENSURE_TRUE(currentInner, NS_OK);
 
@@ -139,6 +143,13 @@
     } break;
   }
 
+  if (nukedOuter) {
+    nsFocusManager* fm = nsFocusManager::GetFocusManager();
+    if (fm) {
+      fm->WasNuked(nukedOuter);
+    }
+  }
+
   return NS_OK;
 }
 